Sources/SwiftCuckoo/Lap.swift

```diff
@@ -0,0 +1,109 @@
+    public enum LapStatus: Equatable {
+        case active                /// Indicates the lap is active.
+        case inactive              /// Indicates the lap is inactive.
+    }
+
+    /// A unique identifier for the lap instance.
+    public var id: Identifier
+
+    /// The start time of the lap.
+    public var startTime: Date?
+
+    /// The optional end time of the lap. Defaults to `nil` if the lap is ongoing.
+    public var endTime: Date?
+
+    /// Initializes a new lap instance with a specified start time and an optional end time.
+    ///
+    /// - Parameters:
+    ///   - startTime: The start time of the lap.
+    ///   - endTime: The end time of the lap. Defaults to `nil` for ongoing laps.
+    public init(id: Identifier, startTime: Date? = nil, endTime: Date? = nil) {
+        self.id = id
+        self.startTime = startTime
+        self.endTime = endTime
+    }
+
+    public mutating func start(on date: Date = Date()) throws {
+        guard lapStatus() == .inactive else {
+            throw Error.lapActive
+        }
+        self.startTime = date
+    }
+
+    /// Stops the lap at the specified date.
+    ///
+    /// - Parameter date: The date when the lap ends. Defaults to the current date.
+    /// - Throws: An error if the lap has already ended.
+    public mutating func stop(on date: Date = Date()) throws {
+        guard lapStatus() == .active else {
+            throw Error.lapNotActive
+        }
+
+        self.endTime = date
+    }
+
+    /// Calculates the duration of the lap.
+    ///
+    /// - Throws: An error if the lap has not ended.
+    /// - Returns: The duration of the lap as a `TimeInterval`.
+    public func duration() throws -> TimeInterval {
+        guard let endTime else {
+            throw Error.lapActive
+        }
+
+        guard let startTime else {
+            throw Error.lapNotActive
+        }
+
+        return endTime.timeIntervalSince(startTime)
+    }
+    
+    /// Determines the current status of a lap.
+    ///
+    /// - Returns: The current status as a `LapStatus` enumerator indicating whether
+    ///   the session is active, or inactive.
+    public func lapStatus() -> LapStatus {
+        guard let endTime else {
+            return .active
+        }
+
+        return .inactive
+    }
+}
+
+// MARK: - Identifier
+
+extension Lap {
+    /// A unique identifier for the session, conforming to `RawRepresentable`,
+    /// `Equatable`, and `Hashable` protocols for easy management and comparison.
+    public struct Identifier: RawRepresentable, Equatable, Hashable {
+        /// The raw string value representing the identifier.
+        public let rawValue: String
+
+        /// Initializes a new identifier with the specified raw value.
+        ///
+        /// - Parameter rawValue: The string value that represents the identifier.
+        public init(rawValue: String) {
+            self.rawValue = rawValue
+        }
+    }
+}
```

Comment:
I think it is unnecessary in that since we can use UUIDs or even a simple index array position to consider a Lap identification.

For example, on Session, we have an identifier to keep it human readable, but on the lap case we just wanna know if session X had Y laps.